rbd/rbd.go

```diff
@@ -119,7 +120,16 @@ func split(buf []byte) (values []string) {
 
 //
 func (e RBDError) Error() string {
-	return fmt.Sprintf("rbd: ret=%d", e)
+	buf := make([]byte, 1024)
+	// strerror expects errno >= 0
+	errno := math.Abs(float64(e))
+
+	ret := C.strerror_r(C.int(errno), (*C.char)(unsafe.Pointer(&buf[0])), C.size_t(len(buf)))
```

Comment:
There seems to be two versions of `strerror_r` and you assume the XSI-compliant version, which only is used if certain macros are set. Otherwise it returns a pointer to the string. Can we rely on that? See man page.

Reply:
I was wondering about that, but it seems Golang sets things up to get the XSI version, and not the GNU one. If you prefer, I can add the needed `#define`s to force the XSI-complaint version.

Reply:
I don't fully understand the difference you are discussing here but if it is not a ton of work to ensure the definition of the function we'll use is reliable I'd say I'm all for it. Worst case we start failing on other platforms when the lib is compiled rather than getting undefined behavior.

Reply:
I'll force using the XSI-complaint version of `strerror_r`. Which version you get depends on the version of glibc, but it can be influenced by `#define`s.